remote-trait-object-tests/src/mod_main/context.rs

```diff
@@ -14,19 +14,43 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use parking_lot::Mutex;
+use once_cell::sync::OnceCell;
 use remote_trait_object::Port;
+use std::fmt;
+use std::fmt::Debug;
 
 pub struct Context {
-    pub cmd_port: Mutex<Option<Port>>,
-    pub ping_port: Mutex<Option<Port>>,
+    inner: OnceCell<ContextInner>,
```

Comment:
Is this because of `impl<T> Sync for OnceCell<T> where T: Send + Sync`?

Reply:
I don't know the point of your question.
The reason why I used OnceCell is these things:

1. What I want to use is the `Arc`. Since Port is Sync, we can use Port's methods with Arc.
2. We must initialize Arc in the constructor. There is a cycle dependency in the initialization process. The handler passed to Port requires Context. Context requires Port. We can't create a Context before Port initialization when we used Arc.
3. We can do a similar thing using RwLock. OnceCell's interface is easier to use than RwLock. OnceCell's semantic is more suitable for this scenario.

Reply:
@majecty Ok, but note that the original tests (updated after removal of global variables) doesn't **own** `Port` in the module context, and that would be more natural in the final interface.